src/hotspot/share/classfile/moduleEntry.cpp

```diff
@@ -466,15 +467,17 @@ void ModuleEntry::iterate_symbols(MetaspaceClosure* closure) {
 }
 
 void ModuleEntry::init_as_archived_entry() {
-  Array<ModuleEntry*>* archived_reads = write_growable_array(_reads);
+  DEBUG_ONLY(
+    if (_reads == nullptr) { _reads_is_growable = true; }
+  )
+  set_archived_reads(write_growable_array(reads()));
```

Comment:
I don't really like this forcing of the flag to make the call to reads() happy.  Maybe something like the
following instead?
```
GrowableArray<ModuleEntry*>* entries = nullptr;
if (_reads != nullptr) { entries = reads(); }
set_archived_reads(write_growable_array(entries));
```
Similarly in load_from_archive with the other flag.

Reply:
The `DEBUG_ONLY` block can be removed?